modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/PersistentPageMemoryStorageEngine.java

```diff
@@ -171,22 +191,36 @@ public CompletableFuture<?> onCreate(ConfigurationNotificationEvent<PersistentPa
                 return nullCompletedFuture();
             }
         });
+
+        // TODO: remove this executor, see https://issues.apache.org/jira/browse/IGNITE-21683
+        destructionExecutor = new ThreadPoolExecutor(
+                0,
+                Runtime.getRuntime().availableProcessors(),
+                100,
```

Comment:
Maybe `0` ?

Reply:
Why?

Reply:
It was like this before, why is this important? These executors will be removed anyway

Reply:
ok